### ISSUE TITLE
CRLF line endings break magics

### ISSUE THREAD

**Issue by kycutler** [ipython/ipython#12471]
When a cell contains a (line or cell) magic followed by a CRLF line ending, the CR character seems to be interpreted as part of the magic name, resulting in an error; e.g. `UsageError: Line magic function '%ls\r' not found.`

### Repro:

crlf.ipynb:
```json
{"cells":[{"cell_type":"code","source":["%ls\r\n"],"outputs":[],"execution_count":null,"metadata":{}}],"metadata":{},"nbformat":4,"nbformat_minor":2}
```

```
$ ipython crlf.ipynb
` not found Line magic function `%ls
```

Notice how the CR character causes the output to "wrap around".

**Comment by captainsafia** [comment]
@Carreau A user reported this issue in the nteract project but it seems like it is best resolved in the ipykernel instead of individually in each front-end.

Do you have any ideas about what might be going on here?

**Comment by Carreau** [comment]
Can try to fix it in IPython itself. We likely have a split(\n) somewhere instead of split lines().

**Comment by kycutler** [comment]
I've done a bit of digging and found these lines which seem to be causing it:

This one affecting line magics: https://github.com/ipython/ipython/blob/58006f3d544616bbd51e2e208b8220899d63eb31/IPython/core/inputtransformer2.py#L156

This one affecting cell magics: https://github.com/ipython/ipython/blob/58006f3d544616bbd51e2e208b8220899d63eb31/IPython/core/inputtransformer2.py#L97

**Comment by Carreau** [comment]
CLosed by #12475
